server/validate.go

```diff
@@ -106,28 +106,33 @@ func validateGetEntryAndProofRequest(req *trillian.GetEntryAndProofRequest) erro
 }
 
 func validateQueueLeafRequest(req *trillian.QueueLeafRequest) error {
-	if req.Leaf == nil {
-		return status.Error(codes.InvalidArgument, "QueueLeafRequest.Leaf empty")
+	if err := validateLogLeaf(req.Leaf); err != nil {
```

Comment:
Could we have a validateLeaf(leaf, "QueueLeaf") function and get rid of some of the duplication? They're all a bit similar.

Reply:
Done. Also did the same in `validateLogLeaves`.